server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -647,9 +654,12 @@ public void updateState(ClusterState state) {
         final Settings previousSettings = metadata.settings();
         metadata = getRepoMetadata(state);
         final Settings updatedSettings = metadata.settings();
-        if (updatedSettings.equals(previousSettings) == false) {
-            snapshotRateLimiter = getRateLimiter(metadata.settings(), MAX_SNAPSHOT_BYTES_PER_SEC);
-            restoreRateLimiter = getRateLimiter(metadata.settings(), MAX_RESTORE_BYTES_PER_SEC);
+        final Settings incomingSettings = state.metadata().settings();
+        boolean recoverySpeedAffected = SETTINGS_AFFECTING_MAX_BYTES_PER_SEC.stream()
```

Comment:
Can you explain why we need this? If the recovery settings change, any new file upload/download will pick up the new recovery settings anyway, only the last files will (maybe) continue on the old rate limiter (and this is only even the case if we flip between no rate limiting and rate limiting). This is how it works today and seems sufficient.

The check here would not be enough anyway, in case someone removes the settings from cluster state, reverting to defaults. But if we want to solve that, it should be solved in `RecoverySettings` by having a rate limiter that does not limit at all rather than null in the unlimited case.

Reply:
Hi @henningandersen ! Indeed, there is no "operational" reason we depend on the recovery settings update here. The only reason we depend on this is due to [this comment](https://github.com/elastic/elasticsearch/pull/91021/files/35c3cc476285f1bb04a3f4b9051fdaff81c9822f#r1017738828), related to the warning we log when the snapshot/restore speed is higher than the effective recovery speed, and we warn that it will be throttled by the lower recovery speed. So we basically want to log a new warning in case the recovery speed is changed and becomes lower than the snapshot/restore speed. We also introduced a relevant test for this warning message (see bottom of SnapshotThrottlingIT.java).

> The check here would not be enough anyway, in case someone removes the settings from cluster state, reverting to defaults.

Oh interesting, I did not think of this. So in this case, incomingSettings would not have any recovery settings? We would miss showing a warning in this case, e.g., if the default recovery speed is less than the snapshot/restore speed. But hopefully this does sound like a blocker for merging the PR. However, it would be interesting to correct this. I believe I should be able to detect this case by comparing the `updatedSettings` vs `incomingSettings`? I can research this a bit and see if I can make a solution in this PR or do a follow-up PR. Feedback is welcome.

Reply:
I'd prefer to keep that case out, i.e., to not get a warning if the recovery settings is lowered explicitly. If we wanted to detect this, we should listen to the recovery settings change, perhaps on the `RecoverySettings` object. Or add a cluster state listener.

But I do not see it as a blocker for merging, it can be handled in a follow-up.

Reply:
Ah OK @henningandersen thanks for  the quick comment! To clarify, when you say

> I'd prefer to keep that case out, i.e., to not get a warning if the recovery settings is lowered explicitly.

Would you like to remove the warning altogether? Or just in the case the recovery settings are removed (set to default), skipping the warning is fine for now?

Reply:
I'd like to keep the warning if you configure or change a repo's bw settings and it is above the recovery settings.

I'd prefer to remove the warning in case you lower the recovery settings, i.e., the half-functioning detection of a change of the cluster settings. So remove the specific line and the check for `|| recoverySpeedAffected` (and fix tests).

Reply:
@henningandersen , I mean, should I change the PR now to remove the warning? Or do it in a follow-up PR?

Reply:
But you can also leave it all for follow-up work if you prefer.

Reply:
Ah I see now @henningandersen , I think I can quickly remove the warning for the case somebody lowers the recovery settings then. Let me see if I can quickly do that and I will try to do it in this PR. Thanks for the feedback!